### ISSUE TITLE
where clause doesn't support nested expressions

### ISSUE THREAD

**Issue by emidln** [jkk/honeysql#135]
It seems llike this should work:

``` clj
(= (honeysql.core/format {:where [:= :a [:/ :b 2]]})
["WHERE a = (b / ?)" 2])
```

Except that we don't  really nest function calls (we use `to-sql`). In an effort to also address  #134 we could attempt to detect if arrays have more than 2 values, and if they do, treat them as function calls. As is, the best we seem to be able to do is something like:

``` clj
{:where [:= :a #sql/call [:/ :a 10]])
```

which seems inconsistent with the brevity afford the top-level expressions in WHERE clauses.

**Comment by seancorfield** [comment]
@emidln I'm looking at this issue and #134 I'm a bit confused because I just tried the `WHERE` clause listed above and it *does* format correctly -- that equality holds. Adding `#sql/call` makes no difference so it looks like this case has been fixed. Can you confirm that whether this is still an issue for you?

**Comment by emidln** [comment]
@seancorfield I can confirm this is fixed now. It looks like it was fixed as of *0.9.0-beta1* fwiw. Thanks for reminding me about this.
